tests/queries/0_stateless/03203_system_query_metric_log.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CURDIR"/../shell_config.sh
+
+readonly query_prefix=$CLICKHOUSE_DATABASE
+
+$CLICKHOUSE_CLIENT --query-id="${query_prefix}_1000" -q "SELECT sleep(2.5) FORMAT Null" &
+$CLICKHOUSE_CLIENT --query-id="${query_prefix}_1234" -q "SELECT sleep(2.5) SETTINGS query_metric_log_interval=1234 FORMAT Null" &
+$CLICKHOUSE_CLIENT --query-id="${query_prefix}_123" -q "SELECT sleep(2.5) SETTINGS query_metric_log_interval=123 FORMAT Null" &
+$CLICKHOUSE_CLIENT --query-id="${query_prefix}_0" -q "SELECT sleep(2.5) SETTINGS query_metric_log_interval=0 FORMAT Null" &
+$CLICKHOUSE_CLIENT --query-id="${query_prefix}_fast" -q "SELECT sleep(0.1) FORMAT Null" &
+
+wait
+
+$CLICKHOUSE_CLIENT -q "SYSTEM FLUSH LOGS"
+
+function check_log()
+{
+    interval=$1
+    # We calculate the diff of each row with its previous row to check whether the intervals at which
+    # data is collected is right. The first row is always skipped because the diff is 0. The same for the
+    # last row, which is skipped because doesn't contain a full interval.
+    $CLICKHOUSE_CLIENT --max_threads=1 -m -q """
+    WITH diff AS (
+        SELECT
+            row_number() OVER () AS row,
+            count() OVER () as total_rows,
+            event_time_microseconds,
+            first_value(event_time_microseconds) OVER (ORDER BY event_time_microseconds ROWS BETWEEN 1 PRECEDING AND 0 FOLLOWING) as prev,
+            dateDiff('ms', prev, event_time_microseconds) AS diff
+        FROM system.query_metric_log
+        WHERE event_date >= yesterday() AND query_id = '${query_prefix}_${interval}'
+        ORDER BY event_time_microseconds
+        OFFSET 1
+    )
+    SELECT if(count() BETWEEN ((ceil(2500 / $interval) - 2) * 0.8) AND ((ceil(2500 / $interval) - 2) * 1.2), 'number_of_metrics_${interval}_ok', 'number_of_metrics_${interval}_error'),
+           if(avg(diff) BETWEEN $interval * 0.8 AND $interval * 1.2, 'timestamp_diff_in_metrics_${interval}_ok', 'timestamp_diff_in_metrics_${interval}_error')
+    FROM diff WHERE row < total_rows
+    """
+}
+
+check_log 1000
+check_log 1234
+check_log 123
+
```

Comment:
It'd be nice to check that we are actually logging something too. Like check that `SleepFunctionCalls` / `SleepFunctionMicroseconds`, `SleepFunctionElapsedMicroseconds` are not 0 (whatever value they must be) in the last / exit log of one of the queries.

Reply:
Yep, I also thought yesterday about checking `ProfileEvent_Query`, `ProfileEvent_SelectQuery` and `ProfileEvent_InitialQuery` are 1 on the first event to ensure we're not losing anything on the first row.

Will do both :+1:

Reply:
Done at [`4fc4cee` (#66532)](https://github.com/ClickHouse/ClickHouse/pull/66532/commits/4fc4cee8627eae133c18d22fcd297d1445922eb8)